### PR TITLE
add a polyfill for HTMLCanvasElement.prototype.toBlob

### PR DESCRIPTION
**Pull request by davidsbailey**
Fixes https://rpm.newrelic.com/accounts/501463/browser/3787364/js_errors#id=-1281773959&tab_index=1&instance_index=0 . This error was introduced in https://github.com/code-dot-org/code-dot-org/pull/14061. The error was caused by calling `HTMLCanvasElement.prototype.toBlob` which is not supported in Safari or IE/Edge.

I reproed this in gamelab on Safari and saw 50+ errors per second since the thumbnail post retries if it does not succeed.

I've verified manually that the PUT to the thumbnail URL is succeeding now in Safari and IE.

This probably warranted a UI test since it affects gamelab even though the thumbnails aren't being exposed publicly in the UI yet. Since the JS error is already in production, I'd like to get this minimal fix in ASAP and then add a UI test in a subsequent PR.

### CODE REVIEW

File: apps/src/utils.js [hunk 1]
```diff
@@ -237,6 +237,26 @@ if (!String.prototype.repeat) {
...
+        arr[i] = binStr.charCodeAt(i);
+      }
+
+      const blob = new Blob([arr], {type: type || 'image/png'});
+      callback(blob);
+    }
+  });
+}
+
+/**
```
Comment (Bjvanminnen) [inline apps/src/utils.js]: Makes sense to put this next to our other polyfill, but wonder if all of our polyfills should be in a separate file from src/utils?

**Comment by islemaster** [comment]
Do we need to double-check that utils is imported anywhere to ensure that the polyfill is always loaded in the contexts where it's needed?

**Comment by davidsbailey** [comment]
@Bjvanminnen @islemaster PTAL

I've extracted polyfills.js from utils.js. I didn't see a great way to ensure polyfills were imported "everywhere", so I imported them explicitly from studioApp figuring that would give us pretty broad coverage.

The UI test failures from [circle/34387](https://circleci.com/gh/code-dot-org/code-dot-org/34387?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) (on this PR) are also failing on staging in [circle/34389](https://circleci.com/gh/code-dot-org/code-dot-org/34389) suggesting they are unrelated to this PR.

**Comment by Bjvanminnen** [comment]
There might be a way to get them included everywhere via webpack. @pcardune might know. Otherwise this lgtm.

**Comment by davidsbailey** [comment]
These are the resources I found on the topic, however none of them seem to do exactly what we want:
* http://webpack.github.io/docs/shimming-modules.html
* https://webpack.js.org/plugins/provide-plugin/

We can probably do the same thing we're already doing with `babel-polyfill` and `idempotent-babel-polyfill` to put these in every entry point and test. However it seems risky, so I'd rather get these new relic errors fixed now, and attempt that in a follow-up PR.

**Comment by Bjvanminnen** [comment]
Im fine with current approach.

**Pull request merged**